apps/gitness/src/App.tsx

```diff
@@ -23,6 +23,14 @@ export default function App() {
           return newRequest
         }
         return request
+      },
+      responseInterceptor: (response: Response) => {
+        switch (response.status) {
```

Comment:
we might need to do `response.clone()` to allow consuming it later in the app (incase we DONT redirect)